HousingSearchApi/Startup.cs

```diff
@@ -159,7 +162,7 @@ private static void ConfigureElasticsearch(IServiceCollection services)
                     .PrettyJson().ThrowExceptions().DisableDirectStreaming();
             var esClient = new ElasticClient(connectionSettings);
 
-            services.AddSingleton<IElasticClient>(esClient);
+            services.TryAddSingleton<IElasticClient>(esClient);
```

Comment:
Does this make much of a difference?

Reply:
It's a lovely trick that someone taught me :)
Basically, by changing it to TryAdd, it doesn't add if it's already been added. And because I am now adding a different implementation in integration tests, it means it doesn't get overwritten during testing !

Reply:
That's cool @evak6979